src/App.tsx

```diff
@@ -1,29 +1,56 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
-import { getNumbers } from './utils';
+import { getNumbers, PER_PAGE } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [selectedPerPageValue, setSelectedPerPageValue] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const handlePerPageChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedPerPageValue(Number(event.target.value));
+    setCurrentPage(1);
+  };
+
+  const onPageChange = (page: number) => {
+    if (page !== currentPage) {
+      setCurrentPage(page);
+    }
+  };
+
+  const indexOfLastItem = currentPage * selectedPerPageValue;
+  const indexOfFirstItem = indexOfLastItem - selectedPerPageValue;
+
+  const itemFrom = indexOfFirstItem + 1;
```

Comment:
The `itemFrom` calculation should start from 0 instead of 1 to correctly index the array. This will ensure that the items are displayed correctly for each page.